### ISSUE TITLE
Error while reading events with an additional string '#aMCatNLO' 

### ISSUE THREAD

**Issue by jhgoh** [scikit-hep/pylhe#102]
**Problem:**
pylhe raises error when I try to read a lhe file generated with aMC@NLO

```
>>> lhe = readLHE("events.lhe")
>>> for x in lhe:
...     print(x)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/store/sw/anaconda3/envs/ds4hep/lib/python3.9/site-packages/pylhe/__init__.py", line 201, in readLHE
    particle_objs = [LHEParticle.fromstring(p) for p in particles]
  File "/store/sw/anaconda3/envs/ds4hep/lib/python3.9/site-packages/pylhe/__init__.py", line 201, in <listcomp>
    particle_objs = [LHEParticle.fromstring(p) for p in particles]
  File "/store/sw/anaconda3/envs/ds4hep/lib/python3.9/site-packages/pylhe/__init__.py", line 71, in fromstring
    return cls(**dict(zip(cls.fieldnames, map(float, string.split()))))
ValueError: could not convert string to float: '#aMCatNLO'
```

The example LHE event block can be seen:
```xml
  <event>
  4      0 0.15466170E+09 0.14090630E+02 0.75467716E-02 0.20024995E+00
       21 -1    0    0  501  502 0.00000000E+00 0.00000000E+00 0.34453202E+03 0.34453202E+03 0.00000000E+00 0.0000E+00 0.9000E+01
       21 -1    0    0  502  503 0.00000000E+00 0.00000000E+00 -.14406924E+00 0.14406924E+00 0.00000000E+00 0.0000E+00 0.9000E+01
        5  1    1    2  501    0 0.41326091E+00 -.45455539E+01 0.23557499E+03 0.23566607E+03 0.47000000E+01 0.0000E+00 0.9000E+01
       -5  1    1    2    0  503 -.41326091E+00 0.45455539E+01 0.10881297E+03 0.10901002E+03 0.47000000E+01 0.0000E+00 0.9000E+01
#aMCatNLO 1  5  2  0  0 0.00000000E+00 0.00000000E+00 9  0  0 0.10000000E+01 0.35498143E+00 0.24370745E+01 0.00000000E+00 0.00000000E+00
  <rwgt>
   <wgt id='1001'> 0.15466E+09 </wgt>
   <wgt id='1002'> 0.24435E+09 </wgt>
   <wgt id='1003'> 0.76688E+08 </wgt>
   <wgt id='1004'> 0.11073E+09 </wgt>
   <wgt id='1005'> 0.17493E+09 </wgt>
   <wgt id='1006'> 0.54902E+08 </wgt>
   <wgt id='1007'> 0.23857E+09 </wgt>
   <wgt id='1008'> 0.37692E+09 </wgt>
   <wgt id='1009'> 0.11829E+09 </wgt>
  </rwgt>
  </event>
```

**My Suggestion**:
Modify the `read_lhe()` function in the `__init__.py`
  - minor fix to strip whitespaces
  - parse LHEParticles not more than the number of particles defined in the 1st line of the event data

```python
def read_lhe(filepath):
    try:
        with _extract_fileobj(filepath) as fileobj:
            for event, element in ET.iterparse(fileobj, events=["end"]):
                if element.tag == "event":
                    data = element.text.strip().split("\n")
                    eventdata, particles = data[0], data[1:]
                    eventinfo = LHEEventInfo.fromstring(eventdata)
                    particles = particles[:int(eventinfo.nparticles)]
                    particle_objs = [LHEParticle.fromstring(p) for p in particles]
                    yield LHEEvent(eventinfo, particle_objs)
    except ET.ParseError as excep:
        print("WARNING. Parse Error:", excep)
        return
```

**Comment by jhgoh** [comment]
@eduardo-rodrigues thank you for your suggestion. I tried other interfaces available in this library and realized there is another function (or class) `read_lhe_with_attributes` which works fine. The difference comes from that the latter one ignores entries starting with '#', but the previous one without attributes just take all particle entries without checking the comment-out strings nor the number of particles from the eventinfo.
The LHE file I tried was nothing new, rather I'll make a PR to fix the issue.
